src/mapml/layers/MapMLLayer.js

```diff
@@ -1136,8 +1162,37 @@ export var MapMLLayer = L.Layer.extend({
                 layer.error = true;
             }
             layer.fire('extentload', layer, false);
+            // update controls if needed based on mapml-viewer controls/controlslist attribute
+            if (layer._layerEl.parentElement) {
+              // if layer does not have a parent Element, do not need to set Controls
+              layer._layerEl.parentElement._toggleControls();
+            }
             layer._layerEl.dispatchEvent(new CustomEvent('extentload', {detail: layer,}));
-        }
+          }
+
+          function _attachToLayer() {
+            let mapml = xhr.responseXML;
+            if (mapml) {
+              let shadowRoot = this._layerEl.shadowRoot || this._layerEl.attachShadow({mode: 'open'});
```

Comment:
I think the attachShadow should be done by the `<layer->` element, either during the constructor (preferable) or the connectedCallback (less preferred, I think, because you then have to check if it exists, whereas if you do it during construction, it will exist from then on).

Reply:
Is it better to make it done in connectedCallback()? Because we need to check attribute "src" to see whether a shadow root is necessary, and attributes of elements are only meaningful AFTER connecting to the DOM tree I believe?

Reply:
Yup if attributes are in play, connectefCallback is the right choice, I believe